internal/release/promote/promotion.go

```diff
@@ -129,10 +131,22 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
 		return "", nil
 	}
 
-	prURL, err := p.perform(list)
+	if opts.TargetEnv.Spec.Promotion.FromPullRequests && !opts.AutoMerge {
```

Comment:
I don't think you lend the proper semantic to `FromPullRequests` property. It should probably be properly commented and documented to make things clearer, but what it really controls is whether `joy pr promote` will consider that environment in the user-prompted list as a potential candidate for promoting a "source code" PR. We will need something else to configure `auto-merge`. Maybe a new `Environment.spec.promotion.autoMergeable` or `.allowAutoMerge` property?